### ISSUE TITLE
node.set_unless does not work

### ISSUE THREAD

**Issue by Jeppesen-io** [chef/chef#5715]
## Description

Very simply, using `node.set_unless['service']['id'] = 6754` in a recipe does not save the attribute to the node, as viewed from `knife node edit <nodename>`

This can been seen when using `chef-zero` and `chef-server` as the backend

## Chef Version
I've tested with the following versions:

`12.8.1`      - Works as expected
`12.9.38`  - Works as expected
`12.10.24` - Works as expected
`12.11.18` -   Works as expected
`12.12.13` - `node.set_unless` does not save attribute
`12.12.15` - `node.set_unless` does not save attribute
`12.16.42` - `node.set_unless` does not save attribute
`12.17.44` - `node.set_unless` does not save attribute

## Platform Version

```lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 14.04.5 LTS
Release:	14.04
Codename:	trusty
```

## Replication Case

I can replicate this easily using `set.node_unless`  with chef-client >= ~~12.12.15~~ 12.12.13

The attribute is not seen via `knife node edit <nodename>`

If I switch to `node.set` or `node.normal_unless`, it works just fine. I only see this with  `node.set_unless`

## Client Output

nothing relevant from `chef-client -l debug`, just the expected deprecation warnings:

``` WARN: node.set_unless is deprecated and will be removed in Chef 14, please use node.default_unless/node.override_unless (or node.normal_unless if you really need persistence) at /opt/chef/embedded/lib/ruby/2.1.0/forwardable.rb:183:in `set_unless' at /opt/chef/embedded/lib/ruby/gems/2.1.0/gems/chef-12.12.15/lib/chef/event_dispatch/dispatcher.rb:43:in `call'```

## Stacktrace

N/A

**Comment by Jeppesen-io** [comment]
Narrowed down the version - `12.12.13` does not work, but `12.11.18` does. 

Perhaps this is related to https://github.com/chef/chef/pull/5097?

**Comment by lamont-granquist** [comment]
probably a dup of #5556?

**Comment by lamont-granquist** [comment]
so node.set_unless was accidentally behaving like node.default_unless not node.normal_unless, plus there was the issue in #5556 -- both of those issues are fixed in #5717 

if that doesn't fix it, then i think this issue needs additional details
